### PR TITLE
BREAKING CHANGE: Drop support for deprecated `--wait` option in `migrate-repo` and `migrate-org`

### PR DESCRIPTION
**Pull request by timrogers**
Up until April, the default behavior of `migrate-repo` and `migrate-org` was to exit immediately after queueing the migration, and not to wait for it to complete. Users could opt in to the "wait" behaviour using the `--wait` option.

In April, we [flipped][1] these defaults around, making waiting the default and adding the `--queue-only` option for the old behavior.

This meant that the `--wait` option could go away - but we wanted to do this in a user-friendly, backwards-compatible way.

At that time, we deprecated the `--wait` option, hiding it from the in-CLI documentation and printing warning when it's used. These warnings have been in place for 3 months, so now is a reasonable time to rip off the band aid and remove the argument.

After this change is released, specifying the old `--wait` option will cause the CLI to error.

Fixes https://github.com/github/gh-gei/issues/1082 and https://github.com/github/gh-gei/issues/1059.

[1]: https://github.com/github/gh-gei/pull/929

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [x] Docs updated (or issue created)
- [ ] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

### CODE REVIEW

File: src/OctoshiftCLI.Tests/ado2gh/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs [hunk 1]
```diff
@@ -240,8 +239,7 @@ public async Task Throws_Decorated_Error_When_Create_Migration_Source_Fails_With
             AdoTeamProject = ADO_TEAM_PROJECT,
             AdoRepo = ADO_REPO,
             GithubOrg = GITHUB_ORG,
-            GithubRepo = GITHUB_REPO,
-            Wait = true,
+            GithubRepo = GITHUB_REPO
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/ado2gh/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs:242]:
```suggestion
            GithubRepo = GITHUB_REPO,
```

File: src/OctoshiftCLI.Tests/ado2gh/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs [hunk 2]
```diff
@@ -267,8 +265,7 @@ public async Task It_Falls_Back_To_Ado_And_Github_Pats_From_Environment_When_Not
...
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/ado2gh/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs:268]:
```suggestion
            GithubRepo = GITHUB_REPO,
```

File: src/OctoshiftCLI.Tests/bbs2gh/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs [hunk 3]
```diff
@@ -527,8 +527,7 @@ public async Task Throws_An_Error_If_Export_Fails()
                 BbsUsername = BBS_USERNAME,
                 BbsPassword = BBS_PASSWORD,
                 BbsProject = BBS_PROJECT,
-                BbsRepo = BBS_REPO,
-                Wait = true
+                BbsRepo = BBS_REPO
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/bbs2gh/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs:530]:
```suggestion
                BbsRepo = BBS_REPO,
```

File: src/OctoshiftCLI.Tests/ado2gh/Commands/MigrateRepo/MigrateRepoCommandTests.cs [hunk 4]
```diff
@@ -33,43 +33,19 @@ public void Should_Have_Options()
...
-                GithubRepo = "gh-repo",
-                Wait = true,
-                AdoPat = adoPat,
-                GithubPat = githubPat,
-            };
-
-            _command.BuildHandler(args, _serviceProvider);
-
-            _mockGithubApiFactory.Verify(m => m.Create(null, githubPat));
-        }
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/ado2gh/Commands/MigrateRepo/MigrateRepoCommandTests.cs:73]: Did you mean to delete this test? It seems like this test should still be relevant.

File: src/OctoshiftCLI.Tests/gei/Commands/MigrateOrg/MigrateOrgCommandArgsTests.cs [hunk 5]
```diff
@@ -82,8 +23,7 @@ public void Source_Pat_Defaults_To_Target_Pat()
                 GithubSourceOrg = SOURCE_ORG,
                 GithubTargetOrg = TARGET_ORG,
                 GithubTargetEnterprise = TARGET_ENTERPRISE,
-                GithubTargetPat = TARGET_PAT,
-                Wait = true,
+                GithubTargetPat = TARGET_PAT
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/gei/Commands/MigrateOrg/MigrateOrgCommandArgsTests.cs:26]:
```suggestion
                GithubTargetPat = TARGET_PAT,
```

File: src/OctoshiftCLI.Tests/gei/Commands/MigrateOrg/MigrateOrgCommandHandlerTests.cs [hunk 6]
```diff
@@ -119,8 +118,7 @@ public async Task Happy_Path_PAT_In_Env()
             {
                 GithubSourceOrg = SOURCE_ORG,
                 GithubTargetOrg = TARGET_ORG,
-                GithubTargetEnterprise = TARGET_ENTERPRISE,
-                Wait = true,
+                GithubTargetEnterprise = TARGET_ENTERPRISE
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/gei/Commands/MigrateOrg/MigrateOrgCommandHandlerTests.cs:121]:
```suggestion
                GithubTargetEnterprise = TARGET_ENTERPRISE,
```

File: src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandArgsTests.cs [hunk 7]
```diff
@@ -26,8 +23,7 @@ public void Defaults_TargetRepo_To_SourceRepo()
             {
                 GithubSourceOrg = SOURCE_ORG,
                 SourceRepo = SOURCE_REPO,
-                GithubTargetOrg = TARGET_ORG,
-                Wait = true
+                GithubTargetOrg = TARGET_ORG
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandArgsTests.cs:26]:
```suggestion
                GithubTargetOrg = TARGET_ORG,
```

File: src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs [hunk 8]
```diff
@@ -313,8 +312,7 @@ public async Task Happy_Path_GithubSource_Ghes()
                 TargetRepo = TARGET_REPO,
                 TargetApiUrl = TARGET_API_URL,
                 GhesApiUrl = GHES_API_URL,
-                AzureStorageConnectionString = AZURE_CONNECTION_STRING,
-                Wait = true
+                AzureStorageConnectionString = AZURE_CONNECTION_STRING
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs:315]:
```suggestion
                AzureStorageConnectionString = AZURE_CONNECTION_STRING,
```

File: src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs [hunk 9]
```diff
@@ -365,8 +363,7 @@ public async Task Github_With_Archive_Urls()
                 TargetRepo = TARGET_REPO,
                 TargetApiUrl = TARGET_API_URL,
                 GitArchiveUrl = gitArchiveUrl,
-                MetadataArchiveUrl = metadataArchiveUrl,
-                Wait = true
+                MetadataArchiveUrl = metadataArchiveUrl
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs:366]:
```suggestion
                MetadataArchiveUrl = metadataArchiveUrl,
```

File: src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs [hunk 10]
```diff
@@ -386,8 +383,7 @@ await FluentActions
                     GithubTargetOrg = TARGET_ORG,
                     TargetRepo = TARGET_REPO,
                     TargetApiUrl = TARGET_API_URL,
-                    GitArchiveUrl = gitArchiveUrl,
-                    Wait = true
+                    GitArchiveUrl = gitArchiveUrl
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs:386]:
```suggestion
                    GitArchiveUrl = gitArchiveUrl,
```

File: src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs [hunk 11]
```diff
@@ -535,8 +530,7 @@ public async Task Ghes_With_NoSslVerify_Uses_NoSsl_Client()
                 TargetApiUrl = TARGET_API_URL,
                 GhesApiUrl = GHES_API_URL,
                 AzureStorageConnectionString = AZURE_CONNECTION_STRING,
-                NoSslVerify = true,
-                Wait = true
+                NoSslVerify = true
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs:533]:
```suggestion
                NoSslVerify = true,
```

File: src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs [hunk 12]
```diff
@@ -717,8 +710,7 @@ public async Task Ghes_Retries_Archive_Generation_On_Any_Error()
                 TargetRepo = TARGET_REPO,
                 TargetApiUrl = TARGET_API_URL,
                 GhesApiUrl = GHES_API_URL,
-                AzureStorageConnectionString = AZURE_CONNECTION_STRING,
-                Wait = true
+                AzureStorageConnectionString = AZURE_CONNECTION_STRING
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs:713]:
```suggestion
                AzureStorageConnectionString = AZURE_CONNECTION_STRING,
```

File: src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs [hunk 13]
```diff
@@ -1182,8 +1174,7 @@ public async Task It_Uses_Aws_If_Arguments_Are_Included()
                 AwsBucketName = awsBucketName,
                 AwsAccessKey = awsAccessKeyId,
                 AwsSecretKey = awsSecretAccessKey,
-                AwsRegion = awsRegion,
-                Wait = true
+                AwsRegion = awsRegion
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/gei/Commands/MigrateRepo/MigrateRepoCommandHandlerTests.cs:1177]:
```suggestion
                AwsRegion = awsRegion,
```

File: RELEASENOTES.md [hunk 14]
```diff
@@ -1 +1,2 @@
-- Fix target repo existence check in GitHub Enterprise Server migrations so it doesn't error if the target repo used to exist, but has been renamed
\ No newline at end of file
+- __BREAKING CHANGE__: Drop support for the deprecated `--wait` option in `migrate-repo` and `migrate-org`. Three months ago, we made waiting for the migration to finish the default behavior and added a new `--queue-only` option. For backwards compatibility reasons, we continued to allow the `--wait` argument and printed warnings if it was specified. From this release onwards, specifying the `--wait` argument will cause the CLI to error.
```
Comment (synthead) [inline RELEASENOTES.md:1]: This is a little wordy for a release note, in my opinion.  How about:

```suggestion
- __BREAKING CHANGE__: Drop deprecated `--wait` option for `migrate-repo` and `migrate-org` commands
```

**Review by synthead — COMMENTED** [review]
Wow this is great!  I have one small suggestion about the release note!

**Review by synthead — APPROVED** [review]
LGTM!

**Pull request merged**